src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -457,14 +462,21 @@ private void Dispose(bool disposing)
             // that the Session is closed before deleting our temporary directory.
             try
             {
+                if (Session != null && Session.graph != IntPtr.Zero)
+                {
+                    Session.graph.Dispose();
+                }
+
                 if (Session != null && Session != IntPtr.Zero)
                 {
                     Session.close(); // invoked Dispose()
                 }
-
-                if (Session != null && Session.graph != IntPtr.Zero)
+            }
+            catch (ObjectDisposedException ex)
+            {
+                using (var channel = _env.Start("TensorflowTransform"))
                 {
-                    Session.graph.Dispose();
+                    channel.Warning($"Caught: {ex.Message} during TensorflowTransform dispose.");
                 }
```

Comment:
:grey_exclamation: Managed resources cannot be used when `disposing` is false. Logging from this location is essentially impossible.